cns/restserver/util.go

```diff
@@ -40,6 +46,49 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SetPnpIDMacaddressMapping() error {
+	service.Lock()
+	defer service.Unlock()
+	VfMacAddressMapping, err := fetchMacAddressPnpIDMapping()
+	if err != nil {
+		return err
+	}
+	service.PnpIDByMacAddress = VfMacAddressMapping
+	return nil
+}
+
+func (service *HTTPRestService) getPNPIDFromMacAddress(macAddress string) (string, error) {
+	if _, ok := service.PnpIDByMacAddress[macAddress]; !ok {
+		if err := service.SetPnpIDMacaddressMapping(); err != nil {
+			return "", err
+		}
+	}
+	return service.PnpIDByMacAddress[macAddress], nil
+}
+
+func fetchMacAddressPnpIDMapping() (map[string]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), Timeout)
+	defer cancel() // The cancel should be deferred so resources are cleaned up
+	cmd := exec.CommandContext(ctx, "powershell", GetMacAddressVFPPnpIDMapping)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]string)
+	if len(output) == 0 {
+		return nil, errors.New("Network adapter not found")
+	}
+	lines := strings.Split(strings.TrimSpace((string(output))), "\n")
+	logger.Printf("received input:%s", lines)
+	for _, line := range lines {
+		parts := strings.Split(line, " ")
+		key := strings.ToUpper(strings.ReplaceAll(parts[0], "-", ":"))
```

Comment:
Any particular reason we make this reformatting?

Reply:
We have seen macaddress formatted with : instead of -

Reply:
This is why @rbtr and I maintain that MAC addresses should be parsed at the application border into an internal type that is purpose-built for dealing with it. See discussion here: https://github.com/Azure/azure-container-networking/pull/2634#discussion_r1530977847

Reply:
+1, stop kicking the can down the road and let's fix these stringly typed variables instead of adding more special handling for them

Reply:
Looked into it. I am using the internal type for parsing, but it is not supported in maps, we will need to use string for maps. If we need to change this in ipconfigs response for CNI, it requires changes in CNI contracts. are you suggesting to go for that?

Reply:
You can write a type to mediate access to the map. This is just a thin wrapper on the map: https://go.dev/play/p/ix2n3cHLJM1

Reply:
I am unable to find a suitable place to add this map, It needs to be present in platform package but then we would not be able to use this in rest server package. Also, We are adding complexity for doing the string conversion, we are storing it as a string in map even using this mediate access. CNS also responds to CNI as string, I do not see any benefit in having this convertor in between. It might have been useful, if CNS had to perform some operations on it.